### ISSUE TITLE
Crash in `contract_executor_t::update_blocking` - This code path is not supposed to use coro_t::wait()

### ISSUE THREAD

**Issue by Tryneus** [rethinkdb/rethinkdb#4670]
Full backtrace:

```
error: Error in ../src/arch/runtime/coroutines.cc at line 252:
error: Assertion failed: [TLS_get_cglobals()->assert_no_coro_waiting_counter == 0] This code path is not supposed to use coro_t::wait().
       Constraint imposed at: ../src/concurrency/watchable.tcc:28
error: Backtrace:
error: Tue Aug 11 16:01:08 2015

       1: rethinkdb_backtrace(void**, int) at rethinkdb_backtrace.cc:100
       2: backtrace_t::backtrace_t() at backtrace.cc:203
       3: lazy_backtrace_formatter_t::lazy_backtrace_formatter_t() at backtrace.cc:283
       4: format_backtrace(bool) at backtrace.cc:198
       5: report_fatal_error(char const*, int, char const*, ...) at errors.cc:83
       6: coro_t::wait() at coroutines.cc:250
       7: cross_thread_mutex_t::co_lock() at cross_thread_mutex.cc:46
       8: cross_thread_mutex_t::acq_t::reset(cross_thread_mutex_t*) at cross_thread_mutex.cc:26
       9: cross_thread_mutex_t::acq_t::acq_t(cross_thread_mutex_t*) at cross_thread_mutex.cc:9
       10: perfmon_collection_t::add(perfmon_membership_t*) at core.cc:84
       11: perfmon_membership_t::perfmon_membership_t(perfmon_collection_t*, perfmon_t*, std::string const&, bool) at core.cc:107
       12: contract_executor_t::execution_wrapper_t::execution_wrapper_t(contract_executor_t*, contract_executor_t::execution_key_t const&, uuid_u const&, uuid_u const&, table_raft_state_t const&) at executor.cc:26
       13: contract_executor_t::execution_wrapper_t::execution_wrapper_t(contract_executor_t*, contract_executor_t::execution_key_t const&, uuid_u const&, uuid_u const&, table_raft_state_t const&) at executor.cc:43
       14: scoped_ptr_t<contract_executor_t::execution_wrapper_t> make_scoped<contract_executor_t::execution_wrapper_t, contract_executor_t*, contract_executor_t::execution_key_t&, uuid_u&, uuid_u const&, table_raft_state_t const&>(contract_executor_t*&&, contract_executor_t::execution_key_t&, uuid_u&, uuid_u const&, table_raft_state_t const&) at scoped.hpp:122
       15: ../build/debug_clang/rethinkdb() [0x1f786df] at 0x1f786df ()
       16: ../build/debug_clang/rethinkdb() [0x1f780e2] at 0x1f780e2 ()
       17: std::function<void (table_raft_state_t const*)>::operator()(table_raft_state_t const*) const at functional:2160
       18: ../build/debug_clang/rethinkdb() [0x1fc8c0e] at 0x1fc8c0e ()
       19: contract_executor_t::update_blocking(signal_t*) at executor.cc:260
       20: std::_Mem_fn<void (contract_executor_t::*)(signal_t*)>::operator()(contract_executor_t*, signal_t*) const at functional:550
       21: void std::_Bind<std::_Mem_fn<void (contract_executor_t::*)(signal_t*)> (contract_executor_t*, std::_Placeholder<1>)>::__call<void, signal_t*&&, 0, 1>(std::tuple<signal_t*&&>&&, std::_Index_tuple<0, 1>) at functional:1145
       22: void std::_Bind<std::_Mem_fn<void (contract_executor_t::*)(signal_t*)> (contract_executor_t*, std::_Placeholder<1>)>::operator()<signal_t*, void>(signal_t*&&) at functional:1203
       23: std::_Function_handler<void (signal_t*), std::_Bind<std::_Mem_fn<void (contract_executor_t::*)(signal_t*)> (contract_executor_t*, std::_Placeholder<1>)> >::_M_invoke(std::_Any_data const&, signal_t*) at functional:1779
       24: std::function<void (signal_t*)>::operator()(signal_t*) const at functional:2160
       25: pump_coro_t::run(auto_drainer_t::lock_t) at pump_coro.cc:90
       26: std::_Mem_fn<void (pump_coro_t::*)(auto_drainer_t::lock_t)>::operator()(pump_coro_t*, auto_drainer_t::lock_t) const at functional:550
       27: void std::_Bind<std::_Mem_fn<void (pump_coro_t::*)(auto_drainer_t::lock_t)> (pump_coro_t*, auto_drainer_t::lock_t)>::__call<void, , 0, 1>(std::tuple<>&&, std::_Index_tuple<0, 1>) at functional:1145
       28: void std::_Bind<std::_Mem_fn<void (pump_coro_t::*)(auto_drainer_t::lock_t)> (pump_coro_t*, auto_drainer_t::lock_t)>::operator()<, void>() at functional:1203
       29: callable_action_instance_t<std::_Bind<std::_Mem_fn<void (pump_coro_t::*)(auto_drainer_t::lock_t)> (pump_coro_t*, auto_drainer_t::lock_t)> >::run_action() at callable_action.hpp:31
       30: callable_action_wrapper_t::run() at runtime_utils.cc:43
       31: coro_t::run() at coroutines.cc:205
       32: coro_t* coro_t::spawn_sometime<std::_Bind<std::_Mem_fn<void (pump_coro_t::*)(auto_drainer_t::lock_t)> (pump_coro_t*, auto_drainer_t::lock_t)> >(std::_Bind<std::_Mem_fn<void (pump_coro_t::*)(auto_drainer_t::lock_t)> (pump_coro_t*, auto_drainer_t::lock_t)>&&) at coroutines.hpp:58
       33: pump_coro_t::notify() at pump_coro.cc:31
       34: contract_executor_t::contract_executor_t(uuid_u const&, mailbox_manager_t*, clone_ptr_t<watchable_t<table_raft_state_t> > const&, watchable_map_t<std::pair<uuid_u, uuid_u>, contract_execution_bcard_t>*, multistore_ptr_t*, base_path_t const&, io_backender_t*, backfill_throttler_t*, backfill_progress_tracker_t*, perfmon_collection_t*) at executor.cc:161
       35: table_manager_t::table_manager_t(uuid_u const&, mailbox_manager_t*, server_config_client_t*, watchable_map_t<std::pair<peer_id_t, uuid_u>, table_manager_bcard_t>*, backfill_throttler_t*, watchable_map_t<std::pair<uuid_u, uuid_u>, empty_value_t>*, base_path_t const&, io_backender_t*, uuid_u const&, multi_table_manager_timestamp_t::epoch_t const&, raft_member_id_t const&, raft_storage_interface_t<table_raft_state_t>*, multistore_ptr_t*, perfmon_collection_t*) at table_manager.cc:45
       36: multi_table_manager_t::active_table_t::active_table_t(multi_table_manager_t*, multi_table_manager_t::table_t*, uuid_u const&, multi_table_manager_timestamp_t::epoch_t const&, raft_member_id_t const&, raft_storage_interface_t<table_raft_state_t>*, multistore_ptr_t*, perfmon_collection_t*) at multi_table_manager.cc:130
       37: scoped_ptr_t<multi_table_manager_t::active_table_t> make_scoped<multi_table_manager_t::active_table_t, multi_table_manager_t*, multi_table_manager_t::table_t*&, uuid_u const&, multi_table_manager_timestamp_t::epoch_t const&, raft_member_id_t const&, raft_storage_interface_t<table_raft_state_t>*&, multistore_ptr_t*, perfmon_collection_t*>(multi_table_manager_t*&&, multi_table_manager_t::table_t*&, uuid_u const&, multi_table_manager_timestamp_t::epoch_t const&, raft_member_id_t const&, raft_storage_interface_t<table_raft_state_t>*&, multistore_ptr_t*&&, perfmon_collection_t*&&) at scoped.hpp:122
       38: multi_table_manager_t::on_action(signal_t*, uuid_u const&, multi_table_manager_timestamp_t const&, multi_table_manager_bcard_t::status_t, boost::optional<table_basic_config_t> const&, boost::optional<raft_member_id_t> const&, boost::optional<raft_persistent_state_t<table_raft_state_t> > const&, mailbox_addr_t<void ()> const&) at multi_table_manager.cc:375
       39: std::_Mem_fn<void (multi_table_manager_t::*)(signal_t*, uuid_u const&, multi_table_manager_timestamp_t const&, multi_table_manager_bcard_t::status_t, boost::optional<table_basic_config_t> const&, boost::optional<raft_member_id_t> const&, boost::optional<raft_persistent_state_t<table_raft_state_t> > const&, mailbox_addr_t<void ()> const&)>::operator()(multi_table_manager_t*, signal_t*, uuid_u const&, multi_table_manager_timestamp_t const&, multi_table_manager_bcard_t::status_t, boost::optional<table_basic_config_t> const&, boost::optional<raft_member_id_t> const&, boost::optional<raft_persistent_state_t<table_raft_state_t> > const&, mailbox_addr_t<void ()> const&) const at functional:550
       40: void std::_Bind<std::_Mem_fn<void (multi_table_manager_t::*)(signal_t*, uuid_u const&, multi_table_manager_timestamp_t const&, multi_table_manager_bcard_t::status_t, boost::optional<table_basic_config_t> const&, boost::optional<raft_member_id_t> const&, boost::optional<raft_persistent_state_t<table_raft_state_t> > const&, mailbox_addr_t<void ()> const&)> (multi_table_manager_t*, std::_Placeholder<1>, std::_Placeholder<2>, std::_Placeholder<3>, std::_Placeholder<4>, std::_Placeholder<5>, std::_Placeholder<6>, std::_Placeholder<7>, std::_Placeholder<8>)>::__call<void, signal_t*&&, uuid_u&&, multi_table_manager_timestamp_t&&, multi_table_manager_bcard_t::status_t&&, boost::optional<table_basic_config_t>&&, boost::optional<raft_member_id_t>&&, boost::optional<raft_persistent_state_t<table_raft_state_t> >&&, mailbox_addr_t<void ()>&&, 0, 1, 2, 3, 4, 5, 6, 7, 8>(std::tuple<signal_t*&&, uuid_u&&, multi_table_manager_timestamp_t&&, multi_table_manager_bcard_t::status_t&&, boost::optional<table_basic_config_t>&&, boost::optional<raft_member_id_t>&&, boost::optional<raft_persistent_state_t<table_raft_state_t> >&&, mailbox_addr_t<void ()>&&>&&, std::_Index_tuple<0, 1, 2, 3, 4, 5, 6, 7, 8>) at functional:1145
       41: void std::_Bind<std::_Mem_fn<void (multi_table_manager_t::*)(signal_t*, uuid_u const&, multi_table_manager_timestamp_t const&, multi_table_manager_bcard_t::status_t, boost::optional<table_basic_config_t> const&, boost::optional<raft_member_id_t> const&, boost::optional<raft_persistent_state_t<table_raft_state_t> > const&, mailbox_addr_t<void ()> const&)> (multi_table_manager_t*, std::_Placeholder<1>, std::_Placeholder<2>, std::_Placeholder<3>, std::_Placeholder<4>, std::_Placeholder<5>, std::_Placeholder<6>, std::_Placeholder<7>, std::_Placeholder<8>)>::operator()<signal_t*, uuid_u, multi_table_manager_timestamp_t, multi_table_manager_bcard_t::status_t, boost::optional<table_basic_config_t>, boost::optional<raft_member_id_t>, boost::optional<raft_persistent_state_t<table_raft_state_t> >, mailbox_addr_t<void ()>, void>(signal_t*&&, uuid_u&&, multi_table_manager_timestamp_t&&, multi_table_manager_bcard_t::status_t&&, boost::optional<table_basic_config_t>&&, boost::optional<raft_member_id_t>&&, boost::optional<raft_persistent_state_t<table_raft_state_t> >&&, mailbox_addr_t<void ()>&&) at functional:1203
       42: std::_Function_handler<void (signal_t*, uuid_u, multi_table_manager_timestamp_t, multi_table_manager_bcard_t::status_t, boost::optional<table_basic_config_t>, boost::optional<raft_member_id_t>, boost::optional<raft_persistent_state_t<table_raft_state_t> >, mailbox_addr_t<void ()>), std::_Bind<std::_Mem_fn<void (multi_table_manager_t::*)(signal_t*, uuid_u const&, multi_table_manager_timestamp_t const&, multi_table_manager_bcard_t::status_t, boost::optional<table_basic_config_t> const&, boost::optional<raft_member_id_t> const&, boost::optional<raft_persistent_state_t<table_raft_state_t> > const&, mailbox_addr_t<void ()> const&)> (multi_table_manager_t*, std::_Placeholder<1>, std::_Placeholder<2>, std::_Placeholder<3>, std::_Placeholder<4>, std::_Placeholder<5>, std::_Placeholder<6>, std::_Placeholder<7>, std::_Placeholder<8>)> >::_M_invoke(std::_Any_data const&, signal_t*, uuid_u, multi_table_manager_timestamp_t, multi_table_manager_bcard_t::status_t, boost::optional<table_basic_config_t>, boost::optional<raft_member_id_t>, boost::optional<raft_persistent_state_t<table_raft_state_t> >, mailbox_addr_t<void ()>) at functional:1779
       43: std::function<void (signal_t*, uuid_u, multi_table_manager_timestamp_t, multi_table_manager_bcard_t::status_t, boost::optional<table_basic_config_t>, boost::optional<raft_member_id_t>, boost::optional<raft_persistent_state_t<table_raft_state_t> >, mailbox_addr_t<void ()>)>::operator()(signal_t*, uuid_u, multi_table_manager_timestamp_t, multi_table_manager_bcard_t::status_t, boost::optional<table_basic_config_t>, boost::optional<raft_member_id_t>, boost::optional<raft_persistent_state_t<table_raft_state_t> >, mailbox_addr_t<void ()>) const at functional:2160
       44: mailbox_t<void (uuid_u, multi_table_manager_timestamp_t, multi_table_manager_bcard_t::status_t, boost::optional<table_basic_config_t>, boost::optional<raft_member_id_t>, boost::optional<raft_persistent_state_t<table_raft_state_t> >, mailbox_addr_t<void ()>)>::read_impl_t::read(read_stream_t*, signal_t*) at typed.hpp:705
       45: mailbox_manager_t::mailbox_read_coroutine(connectivity_cluster_t::connection_t*, auto_drainer_t::lock_t, threadnum_t, unsigned long, std::vector<char, std::allocator<char> >*, long, mailbox_manager_t::force_yield_t) at mailbox.cc:277
       46: ../build/debug_clang/rethinkdb() [0x16a9b7f] at 0x16a9b7f ()
       47: ../build/debug_clang/rethinkdb() [0x16a9a6c] at 0x16a9a6c ()
       48: callable_action_wrapper_t::run() at runtime_utils.cc:43
       49: coro_t::run() at coroutines.cc:205
       50: ../build/debug_clang/rethinkdb() [0x16a935d] at 0x16a935d ()
       51: mailbox_manager_t::on_message(connectivity_cluster_t::connection_t*, auto_drainer_t::lock_t, read_stream_t*) at mailbox.cc:239
       52: connectivity_cluster_t::run_t::handle(keepalive_tcp_conn_stream_t*, boost::optional<peer_id_t>, boost::optional<peer_address_t>, auto_drainer_t::lock_t, bool*) at cluster.cc:1193
       53: connectivity_cluster_t::run_t::on_new_connection(scoped_ptr_t<linux_tcp_conn_descriptor_t> const&, auto_drainer_t::lock_t) at cluster.cc:294
       54: std::_Mem_fn<void (connectivity_cluster_t::run_t::*)(scoped_ptr_t<linux_tcp_conn_descriptor_t> const&, auto_drainer_t::lock_t)>::operator()(connectivity_cluster_t::run_t*, scoped_ptr_t<linux_tcp_conn_descriptor_t> const&, auto_drainer_t::lock_t) const at functional:550
       55: void std::_Bind<std::_Mem_fn<void (connectivity_cluster_t::run_t::*)(scoped_ptr_t<linux_tcp_conn_descriptor_t> const&, auto_drainer_t::lock_t)> (connectivity_cluster_t::run_t*, std::_Placeholder<1>, auto_drainer_t::lock_t)>::__call<void, scoped_ptr_t<linux_tcp_conn_descriptor_t>&, 0, 1, 2>(std::tuple<scoped_ptr_t<linux_tcp_conn_descriptor_t>&>&&, std::_Index_tuple<0, 1, 2>) at functional:1145
       56: void std::_Bind<std::_Mem_fn<void (connectivity_cluster_t::run_t::*)(scoped_ptr_t<linux_tcp_conn_descriptor_t> const&, auto_drainer_t::lock_t)> (connectivity_cluster_t::run_t*, std::_Placeholder<1>, auto_drainer_t::lock_t)>::operator()<scoped_ptr_t<linux_tcp_conn_descriptor_t>&, void>(scoped_ptr_t<linux_tcp_conn_descriptor_t>&) at functional:1203
       57: std::_Function_handler<void (scoped_ptr_t<linux_tcp_conn_descriptor_t>&), std::_Bind<std::_Mem_fn<void (connectivity_cluster_t::run_t::*)(scoped_ptr_t<linux_tcp_conn_descriptor_t> const&, auto_drainer_t::lock_t)> (connectivity_cluster_t::run_t*, std::_Placeholder<1>, auto_drainer_t::lock_t)> >::_M_invoke(std::_Any_data const&, scoped_ptr_t<linux_tcp_conn_descriptor_t>&) at functional:1779
       58: std::function<void (scoped_ptr_t<linux_tcp_conn_descriptor_t>&)>::operator()(scoped_ptr_t<linux_tcp_conn_descriptor_t>&) const at functional:2160
       59: linux_nonthrowing_tcp_listener_t::handle(int) at network.cc:924
       60: std::_Mem_fn<void (linux_nonthrowing_tcp_listener_t::*)(int)>::operator()(linux_nonthrowing_tcp_listener_t*, int) const at functional:550
       61: void std::_Bind<std::_Mem_fn<void (linux_nonthrowing_tcp_listener_t::*)(int)> (linux_nonthrowing_tcp_listener_t*, int)>::__call<void, , 0, 1>(std::tuple<>&&, std::_Index_tuple<0, 1>) at functional:1145
       62: void std::_Bind<std::_Mem_fn<void (linux_nonthrowing_tcp_listener_t::*)(int)> (linux_nonthrowing_tcp_listener_t*, int)>::operator()<, void>() at functional:1203
       63: callable_action_instance_t<std::_Bind<std::_Mem_fn<void (linux_nonthrowing_tcp_listener_t::*)(int)> (linux_nonthrowing_tcp_listener_t*, int)> >::run_action() at callable_action.hpp:31
       64: callable_action_wrapper_t::run() at runtime_utils.cc:43
       65: coro_t::run() at coroutines.cc:205
       66: void coro_t::spawn_now_dangerously<std::_Bind<std::_Mem_fn<void (linux_nonthrowing_tcp_listener_t::*)(int)> (linux_nonthrowing_tcp_listener_t*, int)> >(std::_Bind<std::_Mem_fn<void (linux_nonthrowing_tcp_listener_t::*)(int)> (linux_nonthrowing_tcp_listener_t*, int)>&&) at coroutines.hpp:52
       67: linux_nonthrowing_tcp_listener_t::accept_loop(auto_drainer_t::lock_t) at network.cc:891
       68: std::_Mem_fn<void (linux_nonthrowing_tcp_listener_t::*)(auto_drainer_t::lock_t)>::operator()(linux_nonthrowing_tcp_listener_t*, auto_drainer_t::lock_t) const at functional:550
       69: void std::_Bind<std::_Mem_fn<void (linux_nonthrowing_tcp_listener_t::*)(auto_drainer_t::lock_t)> (linux_nonthrowing_tcp_listener_t*, auto_drainer_t::lock_t)>::__call<void, , 0, 1>(std::tuple<>&&, std::_Index_tuple<0, 1>) at functional:1145
       70: void std::_Bind<std::_Mem_fn<void (linux_nonthrowing_tcp_listener_t::*)(auto_drainer_t::lock_t)> (linux_nonthrowing_tcp_listener_t*, auto_drainer_t::lock_t)>::operator()<, void>() at functional:1203
       71: callable_action_instance_t<std::_Bind<std::_Mem_fn<void (linux_nonthrowing_tcp_listener_t::*)(auto_drainer_t::lock_t)> (linux_nonthrowing_tcp_listener_t*, auto_drainer_t::lock_t)> >::run_action() at callable_action.hpp:31
       72: callable_action_wrapper_t::run() at runtime_utils.cc:43
       73: coro_t::run() at coroutines.cc:205
       74: coro_t* coro_t::spawn_sometime<std::_Bind<std::_Mem_fn<void (linux_nonthrowing_tcp_listener_t::*)(auto_drainer_t::lock_t)> (linux_nonthrowing_tcp_listener_t*, auto_drainer_t::lock_t)> >(std::_Bind<std::_Mem_fn<void (linux_nonthrowing_tcp_listener_t::*)(auto_drainer_t::lock_t)> (linux_nonthrowing_tcp_listener_t*, auto_drainer_t::lock_t)>&&) at coroutines.hpp:58
       75: linux_nonthrowing_tcp_listener_t::begin_listening() at network.cc:684
       76: linux_tcp_listener_t::linux_tcp_listener_t(linux_tcp_bound_socket_t*, std::function<void (scoped_ptr_t<linux_tcp_conn_descriptor_t>&)> const&) at network.cc:966
       77: connectivity_cluster_t::run_t::run_t(connectivity_cluster_t*, std::set<ip_address_t, std::less<ip_address_t>, std::allocator<ip_address_t> > const&, peer_address_t const&, int, int, boost::shared_ptr<semilattice_read_view_t<heartbeat_semilattice_metadata_t> >) at cluster.cc:253
       78: do_serve(io_backender_t*, bool, base_path_t const&, metadata_file_t*, serve_info_t const&, os_signal_cond_t*) at serve.cc:212
       79: serve(io_backender_t*, base_path_t const&, metadata_file_t*, serve_info_t const&, os_signal_cond_t*) at serve.cc:634
       80: run_rethinkdb_serve(base_path_t const&, serve_info_t*, file_direct_io_mode_t, int, boost::optional<boost::optional<unsigned long> > const&, uuid_u const*, server_config_versioned_t const*, cluster_semilattice_metadata_t const*, directory_lock_t*, bool*) at command_line.cc:805
       81: run_rethinkdb_porcelain(base_path_t const&, name_string_t const&, std::set<name_string_t, std::less<name_string_t>, std::allocator<name_string_t> > const&, file_direct_io_mode_t, int, boost::optional<boost::optional<unsigned long> > const&, bool, serve_info_t*, directory_lock_t*, bool*) at command_line.cc:834
       82: void std::_Bind<void (*(base_path_t, name_string_t, std::set<name_string_t, std::less<name_string_t>, std::allocator<name_string_t> >, file_direct_io_mode_t, int, boost::optional<boost::optional<unsigned long> >, bool, serve_info_t*, directory_lock_t*, bool*))(base_path_t const&, name_string_t const&, std::set<name_string_t, std::less<name_string_t>, std::allocator<name_string_t> > const&, file_direct_io_mode_t, int, boost::optional<boost::optional<unsigned long> > const&, bool, serve_info_t*, directory_lock_t*, bool*)>::__call<void, , 0, 1, 2, 3, 4, 5, 6, 7, 8, 9>(std::tuple<>&&, std::_Index_tuple<0, 1, 2, 3, 4, 5, 6, 7, 8, 9>) at functional:1145
       83: void std::_Bind<void (*(base_path_t, name_string_t, std::set<name_string_t, std::less<name_string_t>, std::allocator<name_string_t> >, file_direct_io_mode_t, int, boost::optional<boost::optional<unsigned long> >, bool, serve_info_t*, directory_lock_t*, bool*))(base_path_t const&, name_string_t const&, std::set<name_string_t, std::less<name_string_t>, std::allocator<name_string_t> > const&, file_direct_io_mode_t, int, boost::optional<boost::optional<unsigned long> > const&, bool, serve_info_t*, directory_lock_t*, bool*)>::operator()<, void>() at functional:1203
       84: std::_Function_handler<void (), std::_Bind<void (*(base_path_t, name_string_t, std::set<name_string_t, std::less<name_string_t>, std::allocator<name_string_t> >, file_direct_io_mode_t, int, boost::optional<boost::optional<unsigned long> >, bool, serve_info_t*, directory_lock_t*, bool*))(base_path_t const&, name_string_t const&, std::set<name_string_t, std::less<name_string_t>, std::allocator<name_string_t> > const&, file_direct_io_mode_t, int, boost::optional<boost::optional<unsigned long> > const&, bool, serve_info_t*, directory_lock_t*, bool*)> >::_M_invoke(std::_Any_data const&) at functional:1779
       85: std::function<void ()>::operator()() const at functional:2160
       86: starter_t::run_wrapper(std::function<void ()> const&) at runtime.cc:61
       87: std::_Mem_fn<void (starter_t::*)(std::function<void ()> const&)>::operator()(starter_t*, std::function<void ()> const&) const at functional:550
       88: void std::_Bind<std::_Mem_fn<void (starter_t::*)(std::function<void ()> const&)> (starter_t*, std::function<void ()>)>::__call<void, , 0, 1>(std::tuple<>&&, std::_Index_tuple<0, 1>) at functional:1145
       89: void std::_Bind<std::_Mem_fn<void (starter_t::*)(std::function<void ()> const&)> (starter_t*, std::function<void ()>)>::operator()<, void>() at functional:1203
       90: std::_Function_handler<void (), std::_Bind<std::_Mem_fn<void (starter_t::*)(std::function<void ()> const&)> (starter_t*, std::function<void ()>)> >::_M_invoke(std::_Any_data const&) at functional:1779
       91: std::function<void ()>::operator()() const at functional:2160
       92: callable_action_instance_t<std::function<void ()>&>::run_action() at callable_action.hpp:31
       93: callable_action_wrapper_t::run() at runtime_utils.cc:43
       94: coro_t::run() at coroutines.cc:205
       95: coro_t* coro_t::spawn_sometime<std::function<void ()>&>(std::function<void ()>&) at coroutines.hpp:58
```

**Comment by Tryneus** [comment]
Fix is up in review 3145.

**Comment by Tryneus** [comment]
The fix has been approved and merged into `next` in commit 56e95d0, and cherry-picked into `v2.1.x` in commit f9fc31e.  Will be in release 2.1.1.
